app/src/main/java/com/fabulouszanna/pokedex/model/pokemontypes/PokemonWeaknesses.kt

```diff
@@ -0,0 +1,66 @@
+package com.fabulouszanna.pokedex.model.pokemontypes
+
+import com.fabulouszanna.pokedex.model.PokemonModel
+
+class PokemonWeaknesses(pokemon: PokemonModel) {
+    private val type1 = pokemon.type1
+    private val type2 = pokemon.type2
+
+    private fun getTypeObject(type: String): PokemonType = when (type) {
+        "Bug" -> PokemonType.Bug
+        "Dark" -> PokemonType.Dark
+        "Dragon" -> PokemonType.Dragon
+        "Electric" -> PokemonType.Electric
+        "Fairy" -> PokemonType.Fairy
+        "Fighting" -> PokemonType.Fighting
+        "Fire" -> PokemonType.Fire
+        "Flying" -> PokemonType.Flying
+        "Ghost" -> PokemonType.Ghost
+        "Grass" -> PokemonType.Grass
+        "Ground" -> PokemonType.Ground
+        "Ice" -> PokemonType.Ice
+        "Normal" -> PokemonType.Normal
+        "Poison" -> PokemonType.Poison
+        "Psychic" -> PokemonType.Psychic
+        "Rock" -> PokemonType.Rock
+        "Steel" -> PokemonType.Steel
+        "Water" -> PokemonType.Water
+        else -> throw IllegalArgumentException("Type $type does not exist")
+    }
+
+    private val weaknesses =
+        type2?.let { getTypeObject(type1).weaknesses + getTypeObject(type2).weaknesses }
+            ?: getTypeObject(type1).weaknesses
+    private val resistances =
+        type2?.let { getTypeObject(type1).resistances + getTypeObject(type2).resistances }
+            ?: getTypeObject(type1).resistances
+    private val immunities =
+        type2?.let { getTypeObject(type1).immunities + getTypeObject(type2).immunities }
+            ?: getTypeObject(type1).immunities
+
+    private val weaknessResistanceTable: Map<String, Float> by lazy { getWeaknessTable() }
+
+
+    private fun getWeaknessTable(): Map<String, Float> {
+        val weaknessTable = mutableMapOf<String, Float>()
```

Comment:
This can be written differently. A more immutable way would be with use of `mapOf(List<Pair<*,*>>)` function. See if you can figure it out.

Reply:
I need some more guidance here unfortunately. I tried various things but I can't get my head around it